ewatercycle/analysis/__init__.py

```diff
@@ -0,0 +1,136 @@
+import pandas as pd
+import os
+from typing import Union
+from hydrostats import metrics
+import matplotlib.pyplot as plt
+from matplotlib.dates import DateFormatter
+
+
+def hydrograph(
+    discharge: pd.DataFrame,
+    *,
+    reference: str,
+    precipitation: pd.DataFrame = None,
+    fname: Union[os.PathLike, str] = None,
```

Comment:
I'm not sure if this is necessary. I'm fine with keeping it, but maybe as the last optional argument? Because I think the others are more important. Perhaps also use "save_as" instead of  "fname"? Or just "filename"

Reply:
From your comment I don't understand what is or is not necessary. Can you clarify? Or is this only about the name of filename argument? I like `save_as` tbh.

Reply:
I'm not sure if we need to include a save option in the hydrograph function. My personal preference is to be able to do this myself, so I'd rather have the function return the figure instance. I'm fine with having a save option for convenience, but then I think it should be the last keyword argument.

Reply:
Personal preference aside, having a some sort `filename` argument is what we agreed on when we defined what this function should look like 😅

Reply:
Okay, I missed that one, sorry. Anyway, can we still put it last?